### PR TITLE
[micro-gl] Add a new port

### PR DESCRIPTION
**Pull request by diamant3**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/micro-gl/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "micro-gl",
+  "version-date": "2024-06-18",
+  "description": "Fast, Super Slim, Embeddable, Headers only C++11 vector graphics library, that can run on any 32/64 bits computer without FPU or GPU. No standard library required.",
+  "homepage": "https://micro-gl.github.io/docs/microgl",
+  "license": "MIT"
```
Comment (WangWeiLin-MV) [inline ports/micro-gl/vcpkg.json:6]: According to the [upstream custom license](https://github.com/micro-gl/micro-gl/blob/master/LICENSE.MD), [should be null](https://learn.microsoft.com/en-us/vcpkg/reference/vcpkg-json#license) to indicate that users must read the copyright file.
```suggestion
  "license": ""
```

File: ports/micro-gl/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+}
```
Comment (WangWeiLin-MV) [inline ports/micro-gl/vcpkg.json:7]:
```json
  "dependencies": [
    {
      "name": "vcpkg-cmake",
      "host": true
    },
    {
      "name": "vcpkg-cmake-config",
      "host": true
    }
  ]
```

File: ports/micro-gl/portfile.cmake
Comment (WangWeiLin-MV) [inline ports/micro-gl/portfile.cmake:1]: As a new port, please change all files EOL from CRLF to LF.

File: ports/micro-gl/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,10 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO micro-gl/micro-gl
+    REF 1c1dafeccb1b92467d3fd82de00e022a318c8ce8
+    SHA512 0
```
Comment (WangWeiLin-MV) [inline ports/micro-gl/portfile.cmake:5]: Probably 57dd6d8bd633a79d5837c70572547e090c87bbec3e5e0d4f2914f0af11d683b200a452568a78964a3eaa271a6d316863a9a5f2a5f79ae7649e279904ec950962 not 0

File: ports/micro-gl/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,10 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO micro-gl/micro-gl
+    REF 1c1dafeccb1b92467d3fd82de00e022a318c8ce8
+    SHA512 0
+    HEAD_REF master
+)
+
+file(COPY "${SOURCE_PATH}/include/micro-gl" DESTINATION ${CURRENT_PACKAGES_DIR}/include)
```
Comment (WangWeiLin-MV) [inline ports/micro-gl/portfile.cmake:9]: CMake build is provided in upstream.
```suggestion
vcpkg_replace_string("${SOURCE_PATH}/CMakeLists.txt" "add_subdirectory(examples)" "")

vcpkg_cmake_configure(SOURCE_PATH "${SOURCE_PATH}")

vcpkg_cmake_install()

vcpkg_cmake_config_fixup(PACKAGE_NAME microgl CONFIG_PATH "share/microgl/cmake")
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
```

File: ports/micro-gl/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,10 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO micro-gl/micro-gl
+    REF 1c1dafeccb1b92467d3fd82de00e022a318c8ce8
+    SHA512 0
+    HEAD_REF master
+)
+
+file(COPY "${SOURCE_PATH}/include/micro-gl" DESTINATION ${CURRENT_PACKAGES_DIR}/include)
+file(INSTALL "${SOURCE_PATH}/LICENSE.MD" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```
Comment (WangWeiLin-MV) [inline ports/micro-gl/portfile.cmake:10]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE.MD")
```

File: versions/baseline.json
Comment (WangWeiLin-MV) [inline versions/baseline.json:1]: Generate versions files by `vcpkg x-add-version --all --overwrite-version`

**Review by WangWeiLin-MV — CHANGES_REQUESTED** [review]

**Comment by WangWeiLin-MV** [comment]
Note: I will be converting your PR to draft status. Please click "Ready for review" after making the fix and modifications.

**Review by WangWeiLin-MV — APPROVED** [review]
The port usage tests pass with the following triplets:
* x64-osx
* x64-linux
* x64-windows
* x86-windows

**Comment by BillyONeal** [comment]
I also checked that the installed bits don't stomp on other ports:

image

Thanks for the new port!

**Pull request merged**